### DISCUSSION TITLE
Code graveyard

### DISCUSSION THREAD

**Discussion by cwolock** [statdivlab/rigr#49] [General]
This is where we'll discuss code that can be dropped from the package.

**Comment by cwolock** [comment]
`clusterStatsOld`

**Reply by cwolock** [reply]
Pretty self-explanatory. There's a well-documented version called `clusterStats`. Whether or not we decide to keep that one, `clusterStatsOld` seems superfluous.

**Reply by cwolock** [reply]
I deleted this function. I compared it with `clusterStats` and it has strictly less functionality.

**Comment by cwolock** [comment]
`correlate`

**Reply by cwolock** [reply]
This function computes a correlation matrix for an arbitrary number of variables, optionally within strata. It also provides inference, including t-tests and confidence intervals.

**Comment by cwolock** [comment]
`tabulate`

**Reply by cwolock** [reply]
This function creates a contingency table for categorical variables and computes ORs, RRs, chi-squared statistics, etc. I believe it's similar to the `epi.2by2` function in `epiR` that @jphughes9 used last year in 511

**Reply by jphughes9** [reply]
I'm pretty happy with the epiR and epi.2by2 except it doesn't have the McNemar test ... but then neither does tabulate.

**Reply by cwolock** [reply]
I'll add that this function is mostly a wrapper of `tableStat`, which is itself mostly a wrapper of `descrip`.

**Comment by cwolock** [comment]
`tableStat`

**Reply by cwolock** [reply]
Like `tabulate`, relies heavily on `descrip`. I think (having not really found specific use cases for this function) that it's mostly a wrapper for `descrip`, allowing users to choose subsets of descriptive statistics and providing some sort of nicer formatting.

**Comment by cwolock** [comment]
`scatter`

**Reply by cwolock** [reply]
Basically `plot` with certain defaults (like `jitter`) and superimposed loess/least squares lines.

**Comment by cwolock** [comment]
`oneSample`

**Reply by cwolock** [reply]
This is a big one. It's similar in spirit to `regress`, but for a single variable. Can perform inference for means, proportions, odds, survival probabilities, etc. As far as I can tell, for certain functionality (like a basic t-test) it's redundant with `ttest`, but the output is a bit less detailed. This function certainly seems useful in theory. I'd be curious to hear if/how people use it in practice.

**Comment by cwolock** [comment]
`clusterStats`

**Reply by cwolock** [reply]
Another wrapper for `tableStats` that provides within-cluster descriptive statistics. Also does within-cluster least squares slopes.

**Comment by jphughes9** [comment]
Insofar as 511 is concerned we only do one sample t-tests and proportion tests. ttest() already covers the “mean” option and we talked about developing functions for proportion tests (equivalent of prop.test and binom.test with better output) so I don’t see a need for oneSample from the 511 point of view.

Jim
--------------
Jim Hughes
Professor of Biostatistics
University of Washington           Email: ***@***.******@***.***>
Mailstop 351617                           Phone: 206-616-2721
Seattle, WA  98195                        Fax:      206-616-2724
                                                          Cell:      206-498-2456

**Comment by taylorokonek** [comment]
`lspline` and `lsplineD`

**Reply by taylorokonek** [reply]
We've decided to no longer support spline functionality in `regress`, and all functionality for splines has been removed from `regress`. The functions `lspline` and `lsplineD` still exist however in their own scripts, and we can likely delete them.
